lib/OpenQA/Shared/Plugin/Gru.pm

```diff
@@ -115,7 +116,66 @@ sub is_task_active ($self, $task) {
 # checks if there are worker registered
 sub has_workers ($self) { !!$self->app->minion->backend->list_workers(0, 1)->{total} }
 
+# For some tasks with the same args we don't need to repeat them if they were
+# enqueued less than a minute ago, like 'git fetch'
+sub _find_existing_gru_task ($self, $task, $args, $jobs) {
+    # warn __PACKAGE__ . ':' . __LINE__ . ": ========================== _find_existing_gru_task($task)\n";
+    my $schema = OpenQA::Schema->singleton;
+    $args = [$args] if ref $args eq 'HASH';
+    my $dtf = $schema->storage->datetime_parser;
+    my $dbh = $schema->storage->dbh;
+    my $sql = q{SELECT id, args, created, state, retries, notes, result FROM minion_jobs
+                WHERE state IN ('inactive', 'active', 'finished')
+                AND created >= ? AND task = ? AND args = ? limit 10};
```

Comment:
As you suggested consider to use "array_position" from https://www.postgresql.org/docs/9.5/functions-array.html